moc_openshift.py

```diff
@@ -575,3 +619,60 @@ def update_rolebindings(self, project_name, role, rolebindings_json):
                 payload["metadata"][key] = rolebindings_json["metadata"][key]
         self.logger.debug("payload -> 2: " + json.dumps(payload))
         return self.put_request(url, payload, True)
+
+    # member functions for quotas
+    def generate_openshift_quota(self, project_name, quota_def):
+        payload = {
+            "apiVersion": "v1",
+            "kind": "ResourceQuota",
+            "metadata": {
+                "name": self.generate_quota_name(self, project_name, quota_def),
+                "namespace": project_name,
+            },
+            "spec": {},
+        }
+        for k in quota_def.keys():
+            if k == "scope":
+                payload["spec"]["scopes"] = [quota_def["scope"]]
+            else:
+                payload["spec"][k] = quota_def[k]
+        return payload
+
+    def generate_quota_def(self, payload):
+        scope_count = len(payload["spec"]["scopes"])
+        if scope_count == 0:
+            scope_count = 1
+        quota_def = list({} for i in range(scope_count))
+        for k in payload["spec"].keys():
+            if k == "scopes":
+                for x in range(0, len(payload["spec"]["scopes"])):
+                    quota_def[x]["scope"] = payload["spec"]["scopes"][x]
+            else:
+                for x in range(0, scope_count):
+                    quota_def[x][k] = copy.deepcopy(payload["spec"][k])
```

Comment:
You're calling `copy.deepcopy` but you haven't imported the `copy` module. Running `flake8` in a pre-commit hook will help detect this sort of error before you commit your changes. There are a few others:

```
[lars@madhatter openshift-acct-mgt] (initial_deployment_2)$ flake8 moc_openshift.py
moc_openshift.py:68:30: F541 f-string is missing placeholders
moc_openshift.py:96:22: F821 undefined name 'identities'
moc_openshift.py:99:34: F821 undefined name 'identities'
moc_openshift.py:353:9: F841 local variable 'quota_def' is assigned to but never used
moc_openshift.py:652:39: F821 undefined name 'copy'
moc_openshift.py:671:83: F821 undefined name 'resource_name'
moc_openshift.py:676:9: F841 local variable 'url' is assigned to but never used
moc_openshift.py:676:83: F821 undefined name 'resource_name'
```